p2p/node/api.go

```diff
@@ -116,7 +116,7 @@ func (p *P2PNode) Stop() error {
 	}
 }
 
-func (p *P2PNode) requestFromPeers(topic *pubsubManager.Topic, location common.Location, requestData interface{}, responseDataType interface{}, resultChan chan interface{}) {
+func (p *P2PNode) requestFromPeers(topic *pubsubManager.Topic, requestData interface{}, respDataType interface{}, resultChan chan interface{}) {
```

Comment:
why is this necessary? Isn't there a 1:1 mapping between a request and a response data type? e.g. what `requestData` could we send which has ambiguous response type?

Reply:
Looking at it more, Core would still need to specify responseDataType once at the beginning, and then we should be able to drop it in favor of Topic and then extracting the responseDataType from there whenever we need to use it.